_posts/2025-05-27-openloco-v25.05.md

```diff
@@ -0,0 +1,111 @@
+Well, vanilla Locomotion station objects have their removal costs set to completely fund the
+initial construction costs! This means the removal costs and construction costs zero each other out,
+making the upgrade free. We think this is not the most realistic, but won't be changing this before
+we settle on a new object file format.
+
+
+## Scenario editor improvements
+
+Time doesn't pass in the scenario editor, so you can take your time building leasurely, right?
+Surely, nothing will interfere with this, no? Well, as it turns out, there was a Locomotion bug that
+did just that! @LeftofZen identified a line that wasn't taking the editor into account,
+which made buildings disappear in scenario editor as well. This has now been addressed
+([#3096](https://github.com/OpenLoco/OpenLoco/pull/3096)).
+
+Years ago, OpenLoco introduced a Tile Inspector, which is currently mostly used for debugging purposes.
+This wasn't available in the scenario editor, though. Until this month, thanks to a patch from
+@LeeSpork ([#3097](https://github.com/OpenLoco/OpenLoco/pull/3097))!
+
+
+## Ongoing window system modernisation
+
+Over the last few month's, we've been slowly working to modernise the game's window and widget system.
+This month, we changed windows to render their contents relative to their origins, rather than in
+the full game canvas' screen space ([#2936](https://github.com/OpenLoco/OpenLoco/pull/2936)).
+This actually makes the drawing code much easier to read, hopefully saving us further headaches down the line.
+
+Another changes is the `Window` structs are now passed as references in more places,
+as apposed to raw pointers ([#3103](https://github.com/OpenLoco/OpenLoco/pull/3103)).
+Implicitly, this means `null`-pointers are explicitly avoided, meaning less boilerplate again.
+
+While most of this work is invisible to players, for now, sometimes mistakes slip in. Such is the case
+this month, with a funny bug: the construction window's panel background widget was acting like a button!
+This is fixed in [#3110](https://github.com/OpenLoco/OpenLoco/pull/3110).
+
+
+## Steam install autodetection
+
+As won't come as a surprise to the reader, you need a copy of the original Locomotion game
+to play OpenLoco. Getting set up can be a little tedious if you don't know *exactly* where the original
+game has been installed, however. Thanks to a patch from #LeeSpork, OpenLoco now automatically detects
+Locomotion installs in the default Steam folder on Windows
+([#3098](https://github.com/OpenLoco/OpenLoco/pull/3098)).
+
+
+## Company AI implementation
+
+@duncanspumpkin's efforts to reimplement the company AI functions continue unabated. This month,
+he's started unraveling what is arguably the longest spaghetti yet: the routefinding.
+Not known for its accuracy or directness, the company AI's routefinding can definitely so with
+improvement. Before we can do so, though, it needs to be reimplemented it as-is, in order to fully
```

Comment:
I think you need to remove the `it` here, so it reads "it needs to be reimplemented as-is,"